photutils/detection/findstars.py

```diff
@@ -7,18 +7,20 @@
 import warnings
 import math
 import numpy as np
+import abc
 from astropy.table import Column, Table
 from astropy.utils.exceptions import AstropyUserWarning
 from astropy.stats import gaussian_fwhm_to_sigma
 from .core import _convolve_data, find_peaks
 
 
-__all__ = ['daofind', 'irafstarfind']
+class StarFinder(metaclass=abc.ABCMeta):
+    @abc.abstractmethod
+    def __call__(self, data):
+        """Find potential stars in the given data."""
+        pass
 
-
-def daofind(data, threshold, fwhm, ratio=1.0, theta=0.0, sigma_radius=1.5,
-            sharplo=0.2, sharphi=1.0, roundlo=-1.0, roundhi=1.0, sky=0.0,
-            exclude_border=False):
+class DAOFind(StarFinder):
```

Comment:
It's generally good form to use consistent forms of names.  So instead of ``StarFinder`` and ``DAOFind``, I'd suggest ``DAOFinder`` (or ``DAOStarFinder``).